collab2/foraging/toolkit/utils.py

```diff
@@ -0,0 +1,186 @@
+import math
+from itertools import product
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from scipy.signal import find_peaks
+import warnings
+
+#define a class to streamline object creation 
+class dataObject:
+    def __init__(self, foragersDF, grid_size=None, rewardsDF=None, frames=None):
+        if frames is None:
+            frames = foragersDF["time"].nunique()
+
+        if grid_size is None:
+            grid_size = int(max(max(foragersDF["x"]), max(foragersDF["y"])))
+
+        self.grid_size = grid_size
+        self.num_frames = frames
+        self.foragersDF = foragersDF
+
+        ##PP_TODO : change start at 1 to start at 0
+        if self.foragersDF["forager"].min() == 0:
+            self.foragersDF["forager"] = self.foragersDF["forager"] + 1
+
+        self.foragers = [group for _, group in foragersDF.groupby("forager")]
+
+        if rewardsDF is not None:
+            self.rewardsDF = rewardsDF
+            self.rewards = [group for _, group in rewardsDF.groupby("time")]
+
+        self.num_foragers = len(self.foragers)
+        
+        #raise warning if nan values in DataFrame
+        if foragersDF["x"].isna().any():
+            warnings.warn(f"Nan values in data. Specify handling of missing data using skip_incomplete_frames argument to generate_all_predictors")
+
+        #raise warning if all foragers are not present in any frame 
```

Comment:
this is grammatically ambiguous, please rewrite

Reply:
is this better?